### PR TITLE
Add an extra check to check the providers

### PR DESCRIPTION
**Pull request by sshedi**
tdnf check currently has options to skip conflicts & obsoletes.
However it reports error on providers even though the required packages
are available. This will fix that issue by adding an extra check on
providers.

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>

### CODE REVIEW

File: solv/tdnfpackage.c [hunk 1]
```diff
@@ -1518,134 +1518,162 @@ SolvSplitEvr(
  *      1 : if solver problem type and user specified problem matches
  *      0 : if not matches
  */
-static uint32_t
+static bool
 __should_skip(
```
Comment (ppadmavilasom) [inline solv/tdnfpackage.c]: please follow conventions elsewhere in the codebase for function names.

**Review by ppadmavilasom — COMMENTED** [review]
Looks good. Can you please add a test case for this?

File: solv/tdnfpackage.c [hunk 2]
```diff
@@ -1518,134 +1518,162 @@ SolvSplitEvr(
...
         {
-            dwProblemId = solver_findproblemrule(pSolv, i);
-            type = solver_ruleinfo(
-                       pSolv,
-                       dwProblemId,
-                       &dwSource,&dwTarget,
-                       &dwDep);
-
-            if (__should_skip(type, dwSkipProblem))
+            if (!check_for_providers(pSack, type, pszProblem, prv_pkgname))
```
Comment (ppadmavilasom) [inline solv/tdnfpackage.c:1668]: simple check for providers might satisfy but does it then satisfy the goal? i was thinking about isolating the problem reports and trying to reach the goal independently for each.

Comment (sshedi) [reply solv/tdnfpackage.c:1668]: IMO, satisyfing the goal is a different problem, here user is trying to know is there any problem with the packages, right?

**Comment by sshedi** [comment]
> Looks good. Can you please add a test case for this?

In progress

**Comment by sshedi** [comment]
> > Looks good. Can you please add a test case for this?
> 
> In progress

Done

File: pytests/tests/test_check_skip.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+@pytest.fixture(scope='module', autouse=True)
+def setup_test(utils):
+    yield
+    teardown_test(utils)
+
+def teardown_test(utils):
+    pass
+
+def test_check_skipconflicts(utils):
+    cmd = [ 'tdnf', '--config', '/etc/tdnf/tdnf.conf', 'check', '--skipconflicts']
```
Comment (sidcha) [inline pytests/tests/test_check_skip.py:38]: @sshedi, why are you referring to system config instead of test-repo config here?

**Pull request merged**